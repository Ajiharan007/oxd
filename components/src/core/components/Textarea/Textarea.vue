<template>
  <textarea
    :class="classes"
    :style="style"
    :value="label"
    @focus="onFocus"
    @blur="onBlur"
  />
</template>

<script lang="ts">
import {
  TextareaResize,
  RESIZE_NONE,
  RESIZE_VERTICAL,
  RESIZE_HORIZONTAL,
} from './types';

export default {
  name: 'oxd-textarea',

  props: {
    label: {
      type: String,
      required: true,
    },
    resize: {
      type: String,
      default: RESIZE_VERTICAL,
      validator: function(value: TextareaResize) {
        return (
          [RESIZE_VERTICAL, RESIZE_HORIZONTAL, RESIZE_NONE].indexOf(value) !==
          -1
        );
      },
    },
    style: {
      type: Object,
    },
    error: {
      type: Boolean,
      default: false,
    },
  },

  data() {
    return {
      focused: false,
    };
  },

  computed: {
    classes() {
      return {
        'oxd-textarea': true,
        'oxd-textarea--active': !this.focused,
        'oxd-textarea--focus': this.focused,
        'oxd-textarea--error': this.error,
        [`oxd-textarea--resize-${this.resize}`]: true,
      };
    },
  },

  methods: {
    onFocus(e: Event) {
      this.focused = true;
      this.$emit('focus', e);
    },
    onBlur(e: Event) {
      this.focused = false;
      this.$emit('blur', e);
    },
  },
};
</script>

<style src="./textarea.scss" lang="scss" scoped></style>
