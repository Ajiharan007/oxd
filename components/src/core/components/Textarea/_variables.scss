$oxd-textarea-min-height: 5rem !default;
$oxd-textarea-border--active: $oxd-input-control-border--active;
$oxd-textarea-border--focus: $oxd-input-control-border--focus;
$oxd-textarea-border--error: $oxd-input-control-border--error;
