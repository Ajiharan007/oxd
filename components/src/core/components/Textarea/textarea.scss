@import '../../../styles';
@import 'variables';

.oxd-textarea {
  @include oxd-input-control;
  min-height: $oxd-textarea-min-height;
  scrollbar-color: #cfd3de #f6f5fb;
  scrollbar-width: thin;
}

.oxd-textarea--active {
  border: $oxd-textarea-border--active;
}

.oxd-textarea--focus {
  border: $oxd-textarea-border--focus;
  box-shadow: 1px 1px 6px rgba(35, 35, 36, 0.12);
}

.oxd-textarea--error {
  border: $oxd-textarea-border--error;
  box-shadow: 0 0 5px 0.2rem rgba(223, 9, 16, 0.1);
}

.oxd-textarea--resize-vertical {
  resize: vertical;
}

.oxd-textarea--resize-horizontal {
  resize: horizontal;
}

.oxd-textarea--resize-none {
  resize: none;
}
