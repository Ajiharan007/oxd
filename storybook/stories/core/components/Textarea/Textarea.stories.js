import Textarea from '@orangehrm/oxd/core/components/Textarea/Textarea';

export default {
  title: 'Example/Textarea',
  component: Textarea,
  argTypes: {
    resize: {
      control: {type: 'select', options: ['vertical', 'horizontal', 'none']},
    },
    style: {control: {type: 'object'}},
    error: {control: {type: 'boolean'}},
  },
};

const Template = (args, {argTypes}) => ({
  props: Object.keys(argTypes),
  components: {'oxd-textarea': Textarea},
  template: '<oxd-textarea v-bind="$props" />',
});

export const Default = Template.bind({});
Default.args = {
  label: 'Textarea',
  style: {},
};

export const Colored = Template.bind({});
Colored.args = {
  label: 'Textarea',
  style: {backgroundColor: 'aliceblue'},
};

export const Error = Template.bind({});
Error.args = {
  label: 'Textarea',
  error: true,
};
