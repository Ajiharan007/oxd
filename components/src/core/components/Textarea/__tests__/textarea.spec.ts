import {mount} from '@vue/test-utils';
import Textarea from '@orangehrm/oxd/core/components/Textarea/Textarea.vue';

describe('Textarea.vue', () => {
  it('renders OXD Textarea', () => {
    const label = 'Textarea';
    const wrapper = mount(Textarea, {
      props: {label},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
