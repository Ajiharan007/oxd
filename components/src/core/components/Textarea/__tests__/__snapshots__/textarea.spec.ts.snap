// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea.vue renders OXD Textarea 1`] = `<textarea class="oxd-textarea oxd-textarea--active oxd-textarea--resize-vertical"></textarea>`;
